plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraMetadata.java

```diff
@@ -119,16 +125,31 @@ public CassandraTableHandle getTableHandle(ConnectorSession session, SchemaTable
 
     private static SchemaTableName getTableName(ConnectorTableHandle tableHandle)
     {
-        return ((CassandraTableHandle) tableHandle).getSchemaTableName();
+        return ((CassandraTableHandle) tableHandle).getRequiredNamedRelation().getSchemaTableName();
     }
 
     @Override
     public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         requireNonNull(tableHandle, "tableHandle is null");
+        CassandraTableHandle handle = (CassandraTableHandle) tableHandle;
+        if (handle.getRelationHandle() instanceof CassandraQueryRelationHandle queryRelationHandle) {
+            List<ColumnMetadata> columns = getColumnHandles(queryRelationHandle.getQuery()).stream()
+                    .map(column -> ((CassandraColumnHandle) column).getColumnMetadata())
+                    .collect(toImmutableList());
+            return new ConnectorTableMetadata(getSchemaTableName(handle), columns, ImmutableMap.of(), Optional.empty());
```

Comment:
Why don't we use this ctr `ConnectorTableMetadata(getSchemaTableName(handle), columns)` ?